test/promql/tests/test_network.yaml

```diff
@@ -0,0 +1,123 @@
+# This is the main input for unit testing.
+# Only this file is passed as command line argument.
+
+rule_files:
+  - ../extracted-rules.yaml
+
+evaluation_interval: 30s
+
+tests:
+  - interval: 30s
+    # Series data
+    input_series:
+
+      - series: 'container_network_transmit_bytes_total{namespace="prod", pod="prod-pod"}'
+        values: '0 1 1 1 4 6 _ _ 8 10 _'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="release", namespace="prod", pod="prod-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="prod", pod="pre-prod-pod"}'
+        values: '0 2 2 2 4 5 5 0 4 4 5'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="", namespace="prod", pod="pre-prod-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="test", pod="test-pod"}'
+        values: '0 2 2 2 4 5 5 7 8 9 10'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="test", namespace="test", pod="test-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'container_network_transmit_bytes_total{namespace="another-test", pod="test-pod"}'
+        values: '0 2 2 2 4 5 5 7 8 9 10'
+      - series: 'kube_pod_labels{label_pipelines_appstudio_openshift_io_type="test", namespace="another-test", pod="test-pod"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
+
+
+    promql_expr_test:
+
+      # Test the core metric with namespace=prod
+      # Since the sample at 3m is `_` (no sample), the value we get is the last captured value (6)
+      - expr: appstudio_container_network_transmit_bytes_total{namespace="prod"}
+        eval_time: 3m
+        exp_samples:
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="prod", pod="prod-pod", label_pipelines_appstudio_openshift_io_type="release"}'
+          value: 6
+        - labels: 'appstudio_container_network_transmit_bytes_total{namespace="prod", pod="pre-prod-pod"}'
+          value: 5
+
+      # Test the core metric with namespace=test
```

Comment:
I'll write the namespace name with quotations for easier CTRL+F ('# Test the core metric with "namespace=test"'
(same for all other comments)

Reply:
Done